webapp/src/js/enketo/widgets/phone-widget.js

```diff
@@ -5,38 +5,48 @@ const $ = require( 'jquery' );
 const phoneNumber = require('@medic/phone-number');
 require( 'enketo-core/src/js/plugins' );
 
-// Set up enketo validation for `phone` input type
-FormModel.prototype.types.tel = {
-  validate: function( fieldValue ) {
-    return window.CHTCore.Settings
-      .get()
-      .then( function( settings ) {
-        if ( !phoneNumber.validate( settings, fieldValue ) ) {
-          throw new Error( 'invalid phone number: "' + fieldValue + '"' );
-        }
-        return phoneNumber.normalize( settings, fieldValue );
-      } )
-      .then( function( phoneNumber ) {
-        // Check the phone number is unique.  N.B. this makes the
-        // assumption that we have an object type `person` with a
-        // field `phone`.
+const isContactPhoneValid = (settings, fieldValue) => {
+  if (phoneNumber.validate(settings, fieldValue)) {
+    return true;
+  }
 
-        const DB = window.CHTCore.DB;
-        return DB.get().query('medic-client/contacts_by_phone', { key: phoneNumber });
-      } )
-      .then( function( res ) {
-        if ( res.rows.length === 0 ) {
-          return true;
-        }
+  console.error( 'invalid phone number: "' + fieldValue + '"' );
```

Comment:
Good question.  Yes, this is logging to the console in prod.  The tricky part here is that currently there is no way without debugging into the code to know if the provided phone number validation failed because it is not a proper phone number for the current locale or if it is a duplicate phone number.  On the front-end, the app just shows a constraint failure, but the user does not know which it is.  The plan is to address this issue in https://github.com/medic/cht-core/issues/7514.  

With the previous implementation of this validation logic where an exception was thrown instead of returning `false`, the Enekto code would [log the exception message](https://github.com/enketo/enketo/blob/534bf69ef84a56df0f41bfffd69401ee25439284/packages/enketo-core/src/js/form.js#L1442) to `console.error`. So, I am essentially trying to re-create that behavior here.  Printing the error to the console is not really helpful to the user, but it could be useful to an app-dev or someone else trying to troubleshoot unexpected behavior.  Ideally, we can remove this logging when https://github.com/medic/cht-core/issues/7514 is addressed.

Reply:
Cool, that makes sense.